TKGame/Animations/ScreenTransistion.cs

```diff
@@ -0,0 +1,108 @@
+        private bool loop = true;
+
+
+
+        #endregion
+        /// <summary>
+        /// creates ScreenTransition Object
+        /// </summary>
+        /// <param name="graphicsDevice"></param>
+        public ScreenTransition (GraphicsDevice graphicsDevice)
+        {
+            //individual sprite images
+            sourceRectangles = new Rectangle[16];
+            sourceRectangles[1] = new Rectangle(buffer, buffer, spriteX, spriteY);
+            sourceRectangles[2] = new Rectangle(spriteX +(buffer*2), buffer, spriteX, spriteY);
+            sourceRectangles[3] = new Rectangle(buffer, spriteY + (buffer * 2), spriteX, spriteY);
+            sourceRectangles[4] = new Rectangle(spriteX + (buffer*2), spriteY + (buffer * 2), spriteX, spriteY);
+            sourceRectangles[5] = new Rectangle((spriteX  * 2) + (buffer * 2), spriteY + (buffer * 2), spriteX, spriteY);
+            sourceRectangles[6] = new Rectangle(buffer, (spriteY *2) + (buffer), spriteX, spriteY);
+            sourceRectangles[7] = new Rectangle((spriteX + buffer) + 2, (spriteY + (buffer * 2)) * 2, spriteX, spriteY);
+            sourceRectangles[8] = new Rectangle((spriteX * 2) + (buffer * 2), (spriteY + (buffer * 2)) * 2, spriteX, spriteY);
+            sourceRectangles[9] = new Rectangle(buffer, (spriteY * 3) + (2 * buffer) , spriteX, spriteY);
+            sourceRectangles[10] = new Rectangle((spriteX + buffer) + 2, (spriteY * 3) + (2 * buffer), spriteX, spriteY);
+            sourceRectangles[11] = new Rectangle((spriteX * 2) + (buffer * 2), (spriteY * 3) + (6 * buffer), spriteX, spriteY);
+            sourceRectangles[12] = new Rectangle(buffer, (spriteY + buffer) * 4, spriteX, spriteY);
+            sourceRectangles[13] = new Rectangle((spriteX + buffer) + 2, (spriteY + buffer) * 4, spriteX, spriteY);
+            sourceRectangles[14] = new Rectangle((spriteX * 2) + (buffer * 2), (spriteY + buffer) * 4, spriteX, spriteY);
+            sourceRectangles[15] = new Rectangle(buffer, (spriteY + buffer) * 5, spriteX, spriteY);
+            sourceRectangles[0] = new Rectangle((spriteX + buffer) + 2, (spriteY +buffer) * 5 + 16, spriteX, spriteY);
+
+        }
+        /// <summary>
+        /// Draws Screen Transition Rectangle the size of game window
+        /// </summary>
+        /// <param name="spriteBatch"></param>
+        public void Draw(SpriteBatch spriteBatch)
+        {
+            //Draw a rectangle the size of background
+            spriteBatch.Draw(Art.LoadTexture, new Rectangle(0, 0, screenWidth, screenHeight), sourceRectangles[currentIndex], Color.White);
+        }
+
+        /// <summary>
+        /// Updates transition spritesheet through different images 
+        /// </summary>
+        /// <param name="gameTime"></param>
+        public void Update(GameTime gameTime)
+        {
+            if (timer > threshold + 500) //waits for about half a second and then resets the loop 
+                loop = true;
+
+                if (!loop)
```

Comment:
This will always be false once in this if condition because of line 80

Reply:
it is only meant to stop the loop of the animation immediately after one full loop, that is why there is such a big threshold on line 79.